lib/piper/permissions/ast/rule.ex

```diff
@@ -31,4 +31,30 @@ defmodule Piper.Permissions.Ast.Rule do
   end
   def find_command_name(_), do: nil
 
+  def permissions_used(rule = %__MODULE__{}) do
+    Enum.sort(permissions_used(rule.permission_selector, []))
+  end
+
+  def permissions_used(nil, accum) do
+    accum
+  end
+  def permissions_used(%Ast.PermissionExpr{perms: %Ast.List{values: values}}, accum) do
+    Enum.reduce(values, accum, fn(v, acc) ->
+      if Enum.member?(acc, v.value) do
```

Comment:
Using `Enum.uniq` or better yet `MapSet` + `MapSet.put` might clean this stuff up a bit.